service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java

```diff
@@ -79,6 +87,34 @@ public UserDataResponse getUserData(JWTData userData) {
     return new UserDataResponse(user.getFirstName(), user.getLastName(), user.getEmail());
   }
 
+  @Override
+  public UserTeamsResponse getUserTeams(JWTData userData) {
+    int userId = userData.getUserId();
+    UsersRecord user = db.selectFrom(USERS).where(USERS.ID.eq(userId)).fetchOne();
+
+    if (user == null) {
+      throw new UserDoesNotExistException(userData.getUserId());
+    }
+
+    Result<Record2<String, Integer>> teams =
+        db.select(TEAMS.TEAM_NAME, TEAMS.ID)
+            .from(USERS_TEAMS)
+            .join(TEAMS)
+            .onKey()
+            .where(USERS_TEAMS.USER_ID.eq(userId))
```

Comment:
Don't forget to also filter teams where their status is None or Pending

Reply:
Good catch, fixed